aiida_quantumespresso/parsers/raw_parser_pw.py

```diff
@@ -1698,6 +1704,57 @@ def parse_pw_text_output(data, xml_data={}, structure_data={}, input_dict={}, pa
                 except Exception:
                     parsed_data['warnings'].append('Error while parsing stress tensor.')
 
+            # Electronic and ionic dipoles when 'lelfield' was set to True in input parameters
+            elif lelfield is True:
+
+                if 'Electronic Dipole per cell' in line:
+                    electronic_dipole = float(line.split()[-1])
+                    trajectory_frame.setdefault('electronic_dipole_cell_average', []).append(electronic_dipole)
+
+                elif 'Ionic Dipole per cell' in line:
+                    ionic_dipole = float(line.split()[-1])
+                    trajectory_frame.setdefault('ionic_dipole_cell_average', []).append(ionic_dipole)
+
+                elif 'Electronic Dipole on Cartesian axes' in line:
+                    electronic_dipole = [float(data_step[count + i + 1].split()[1]) for i in range(3)]
+                    trajectory_frame.setdefault('electronic_dipole_cartesian_axes', []).append(electronic_dipole)
+
+                elif 'Ionic Dipole on Cartesian axes' in line:
+                    ionic_dipole = [float(data_step[count + i + 1].split()[1]) for i in range(3)]
+                    trajectory_frame.setdefault('ionic_dipole_cartesian_axes', []).append(ionic_dipole)
+
+
+        # End of trajectory frame, only keep last entries for dipole related values
+        if lelfield is True:
+
+            # For every property only get the last entry if possible
+            try:
+                ed_cell = trajectory_frame['electronic_dipole_cell_average'].pop()
+            except IndexError:
+                ed_cell = None
+
+            try:
+                ed_axes = trajectory_frame['electronic_dipole_cartesian_axes'].pop()
+            except IndexError:
+                ed_axes = None
+
+            try:
+                id_cell = trajectory_frame['ionic_dipole_cell_average'].pop()
+            except IndexError:
+                id_cell = None
+
+            try:
+                id_axes = trajectory_frame['ionic_dipole_cartesian_axes'].pop()
+            except IndexError:
+                id_axes = None
+
+            # Only add them if all four properties were successfully parsed
+            if any([value is None for value in [ed_cell, ed_axes, id_cell, id_axes]]):
```

Comment:
Sorry to bother again, like this it is going to append *only* if at least one is *not* found, meaning that if they are all found, it's not going to append it, right?
Maybe you can just remove the if, and the missing values get set to None? Also because anyway you popped out some values, so if you don't put them back somewhere, you would lose it?
Or do you have different idea in mind?